### PR TITLE
Ensure String literals that are only used as classids are initialised

### PR DESCRIPTION
**Pull request by smowton**
Previously they would only get initialised if they happened to be used elsewhere as well,
which of course usually happened in regression tests.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 0ef977cd934ac90f3986e3bc482200113ea8975f).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/101113527

File: jbmc/src/java_bytecode/java_entry_point.cpp [hunk 1]
```diff
@@ -154,8 +155,14 @@ static void java_static_lifetime_init(
   for(const auto &entry : symbol_table.symbols)
     symbol_names.push_back(entry.first);
 
-  for(const auto &symname : symbol_names)
+  // Don't use a for-each loop here because the loop extends the list, and the
```
Comment (thk123) [inline jbmc/src/java_bytecode/java_entry_point.cpp:158]: :bulb: code might read more naturally recursively.

**Review by thk123 — COMMENTED** [review]
Change looks fine, I don't really understand how the test demonstrates it, isn't this a case where the `classids` are being used elsewhere?

**Comment by smowton** [comment]
The important thing is I don't use `"mypackage.MyClass"` somewhere else, i.e. the literal is created solely for the Class's `name` field.

**Review by thk123 — APPROVED** [review]

**Pull request merged**